coins/monero/src/wallet/seed/mod.rs

```diff
@@ -67,26 +67,52 @@ impl Seed {
       CLASSIC_SEED_LENGTH | CLASSIC_SEED_LENGTH_WITH_CHECKSUM => {
         ClassicSeed::from_string(words).map(Seed::Classic)
       }
+      POLYSEED_LENGTH => PolySeed::from_string(words).map(Seed::PolySeed),
       _ => Err(SeedError::InvalidSeedLength)?,
     }
   }
 
   /// Create a Seed from entropy.
-  pub fn from_entropy(lang: Language, entropy: Zeroizing<[u8; 32]>) -> Option<Seed> {
-    ClassicSeed::from_entropy(lang, entropy).map(Seed::Classic)
+  pub fn from_entropy(seed_type: SeedType, entropy: Zeroizing<[u8; 32]>) -> Option<Seed> {
+    match seed_type {
+      SeedType::Classic(lang) => ClassicSeed::from_entropy(lang, entropy).map(Seed::Classic),
+      SeedType::PolySeed(lang) => {
+        let birthday = SystemTime::now().duration_since(UNIX_EPOCH).unwrap().as_secs();
+        let data = PolySeedData { birthday, features: 0, entropy, checksum: 0 };
+        PolySeed::from_polyseed_data(data, lang).map(Seed::PolySeed)
+      }
+    }
+  }
+
+  /// Create a Seed from given data.
+  pub fn from_polyseed_data(data: PolySeedData, lang: polyseed::Language) -> Option<Seed> {
+    PolySeed::from_polyseed_data(data, lang).map(Seed::PolySeed)
```

Comment:
Is it safe for the user to provide the checksum with this function?

Reply:
I think so because checksum is ignored and always overwrited according to other field values.